### DISCUSSION TITLE
Importing multiple tables from a compressed (e.g., zip, gzip) file

### DISCUSSION THREAD

**Discussion by ajfriend** [duckdb/duckdb#9352] [Ideas]
I have a use case where I'd like to share datasets that are naturally represented as multiple tables (a database), and it would be convenient to share the dataset as a single file (e.g., zip, gzip, or something else), but it would also be nice to easily import this data into DuckDB.

For example, each dataset might represent the purchases from customers in a single city on a certain day, represented by 3 tables: `orders`, `customers`, `merchants`.

Right now, I create a parquet file for each table, and group them in a file like `2023-10-15_nyc.zip`. (But in general, they might be `.csv`, `.feather`, `.json`, etc. for each table.)

It would be nice if there were syntax/semantics like:

```sql
SELECT * FROM 'orders.parquet@2023-10-15_nyc.zip'
```

Currently, I have to unzip the file first before reading into DuckDB. I'm working in Python so I can use the standard library to do this, but it would be very ergonomic to have DuckDB "read through" the zip file for me. 

Would something like this make sense?

Alternatively, is there a better way to share datasets that consist of multiple tables (databases) than what I'm doing? I could save a `.duckdb` file representing the database, but I'd like something more portable, as not all consumers would use DuckDB. I could export parquet files to a directory, like in https://duckdb.org/docs/sql/statements/export.html, but a single zip file is easier to share than a directory. 

So this is either a feature request or a request for someone to direct me to a better way to share databases as single files. :)

**Comment by Mause** [comment]
You can probably use fsspec for this, but obviously there will be a performance cost

**Reply by ajfriend** [reply]
Thanks. I wasn't aware of `fsspec`, so I'll take a look! But it would also be nice to have a solution that isn't Python-specific.

Put another way, it seems like we have many interoperable data formats for storing tables (csv, parquet, feather, etc.), but maybe not for databases (multiple tables)? Is it just the case that no such open format exists?

**Comment by ajfriend** [comment]
Looking around, it seems like the SQLite interface comes the closest, with the `sqlite_scan` and `sqlite_attach` functions: https://duckdb.org/docs/guides/import/query_sqlite

Is that currently the best/only way to do what I'm trying to describe here?

And not knowing a ton about SQLite, I'm also wondering if that interface would also support the kinds of compression that I can get with a format like parquet.
